src/__tests__/extensions/surveys.test.ts

```diff
@@ -31,28 +40,324 @@ describe('survey display logic', () => {
+        expect(result.current.isPopupVisible).toBe(false)
+        act(() => {
+            jest.advanceTimersByTime(500)
+        })
+        expect(result.current.isPopupVisible).toBe(true)
+        jest.useRealTimers()
+    })
+
+    test('should not throw an error if posthog is undefined', () => {
+        const { result } = renderHook(() => usePopupVisibility(mockSurvey, undefined, 0, false, removeSurvey))
+        expect(result.current.isPopupVisible).toBe(true)
+    })
+
+    test('should clean up event listeners on unmount when delay is 0', () => {
+        const { unmount } = renderHook(() => usePopupVisibility(mockSurvey, mockPostHog, 0, false, removeSurvey))
+        const removeEventListenerSpy = jest.spyOn(window, 'removeEventListener')
+
+        unmount()
+
+        expect(removeEventListenerSpy).toHaveBeenCalledWith('PHSurveyClosed', expect.any(Function))
+        expect(removeEventListenerSpy).toHaveBeenCalledWith('PHSurveySent', expect.any(Function))
+    })
+
+    test('should dispatch PHSurveyShown event when survey is shown', () => {
+        const dispatchEventSpy = jest.spyOn(window, 'dispatchEvent')
+        renderHook(() => usePopupVisibility(mockSurvey, mockPostHog, 0, false, removeSurvey))
+
+        expect(dispatchEventSpy).toHaveBeenCalledWith(new Event('PHSurveyShown'))
+    })
+
+    test('should handle multiple surveys with overlapping conditions', () => {
+        jest.useFakeTimers()
+        const mockSurvey2 = { ...mockSurvey, id: 'testSurvey2', name: 'Test survey 2' } as Survey
+        const { result: result1 } = renderHook(() =>
+            usePopupVisibility(mockSurvey, mockPostHog, 0, false, removeSurvey)
+        )
+        const { result: result2 } = renderHook(() =>
+            usePopupVisibility(mockSurvey2, mockPostHog, 500, false, removeSurvey)
+        )
+
+        expect(result1.current.isPopupVisible).toBe(true)
+        expect(result2.current.isPopupVisible).toBe(false)
+
+        act(() => {
+            jest.advanceTimersByTime(500)
+        })
+
+        expect(result2.current.isPopupVisible).toBe(true)
+        jest.useRealTimers()
+    })
+})
+
+describe('SurveyManager', () => {
+    let mockPostHog: PostHog
+    let surveyManager: SurveyManager
+    let mockSurveys: Survey[]
+
+    beforeEach(() => {
+        mockPostHog = {
+            getActiveMatchingSurveys: jest.fn(),
+            get_session_replay_url: jest.fn(),
+            capture: jest.fn(),
+        } as unknown as PostHog
+
+        surveyManager = new SurveyManager(mockPostHog)
+
+        mockSurveys = [
+            {
+                id: 'survey1',
+                name: 'Survey 1',
+                type: SurveyType.Popover,
+                appearance: { surveyPopupDelay: 0 },
+                questions: [],
+                conditions: {},
+                start_date: '2021-01-01T00:00:00.000Z',
+                end_date: null,
+                targeting_flag_key: null,
+            },
+        ] as unknown as Survey[]
```

Comment:
Ha, I was just being a bit lazy.  FWIW, the actual values of the survey data don't matter than much in this specific test, and I noticed that some of the existing test code were also using type coercions to handle the fact that these test surveys didn't compile (since they no longer match the `Survey` type).  I can clean all these test surveys up and make them match the correct schema; it's not much extra work and may save us in the future, but for these examples it didn't matter much since I'm not strictly testing the values within these surveys – I just needed some data I could pass in!

Anyway, I'll clean these up.  It's a fair callout

Reply:
Ah all good, upto you, no strong feelings from me here 🤷